### ISSUE TITLE
Unable to propose a proposal with an `approve_proposal` operation

### ISSUE THREAD

**Issue by vikramrajkumar** [bitshares/bitshares-core#214]
_From @abitmore on June 4, 2016 19:0_

Got this message while trying to propose a proposal with an `approve_proposal` operation:

```
10 assert_exception: Assert Exception
other.size() == 0:
    {}
    th_a  proposal_evaluator.cpp:54 do_evaluate
```

The issue is probably caused by lack of size checking on `auth.key_auths` before [this line in proposal.cpp](https://github.com/cryptonomex/graphene/blob/2.0.160328/libraries/chain/protocol/proposal.cpp#L92), imo the code could be

```
if( auth.key_auths.size() > 0 )
   o.emplace_back( std::move(auth) );
```

To fix this issue, a hard fork is needed.

_Copied from original issue: cryptonomex/graphene#658_

**Comment by vikramrajkumar** [comment]
_From @abitmore on January 3, 2017 22:59_

Actually it should be `proposal_update_operation` (and if to add an account authority to "approve" the proposal).

**Comment by xiangxn** [comment]
FC_ASSERT( other.size() == 0)
This should be:'op' with special permission requirements, cannot create a proposal.
Because these ‘op’s may not be able to execute because of missing signatures

**Comment by pmconrad** [comment]
Reopened - replay is failing with pr #841

**Comment by pmconrad** [comment]
#841 contained a fix, plus hardfork protection to prevent a proposal_update from being included in a proposal_create.
It was overlooked, however, that while a proposal_create(proposal_update) currently fails to execute and therefore can't be integrated into the blockchain, a proposal_create(proposal_create(proposal_update)), i. e. a proposal_update nested in two layers of proposal_create, does not currently fail, and has in fact already appeared on-chain.

**Comment by abitmore** [comment]
I was thinking of the same thing.. Thank you @pmconrad for the explanation.

**Comment by abitmore** [comment]
Fixed with #857.
